probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+        self.fitted = True
+        return self
+
+    def _calculate_results(self, X, y, clf, strategy):
+        """
+        Method to calculate the results for a particular imputation strategy.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            clf (binary classifier,sklearn.Pipeline):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategy(string):
+                Name of the strategy used for imputation.
+
+        Returns:
+
+            temp_df(dict) :  Dictionary containing the results of the evaluation.
+        """
+        imputation_results = cross_val_score(
+            clf, X, y, scoring=self.scorer.scorer, cv=self.cv, n_jobs=self.n_jobs
+        )
+
+        temp_results = {
+            "strategy": strategy,
+            f"{self.scorer.metric_name} score": np.round(
+                np.mean(imputation_results), 3
+            ),
+            f"{self.scorer.metric_name} std": np.round(np.std(imputation_results), 3),
+        }
+
+        return temp_results
+
+    def compute(self):
+        """
+        Checks if fit() method has been run and computes the DataFrame with results of imputation for each
+        strategy.
+
+        Returns:
+            (pd.DataFrame):
+                 DataFrame with results of imputation for each strategy.
+        """
+        self._check_if_fitted()
+        return self.report_df
+
+    def fit_compute(self, X, y, column_names=None):
+        """
+        Calculates the cross validated results for various imputation strategies.
+
+        Args:
+            X (pd.DataFrame):
+                input variables.
+
+            y (pd.Series):
+                target variable.
+
+            column_names (None, or list of str, optional):
+                List of feature names for the dataset.
+                If None, then column names from the X dataframe are used.
+
+        Returns:
+            (pd.DataFrame):
+                DataFrame with results of imputation for each strategy.
+
+        """
+        self.fit(X, y, column_names=column_names)
+        return self.compute()
+
+    def plot(self, show=True, **figure_kwargs):
+        """
+        Generates plot of the performance of various imputation strategies.
+
+        Args:
+            show (bool, optional):
+                If True, the plots are showed to the user, otherwise they are not shown. Not showing plot can be useful,
+                when you want to edit the returned axis, before showing it.
+
+            **figure_kwargs:
+                Keyword arguments that are passed to the plt.figure, at its initialization.
+
+        Returns:
+            (plt.axis):
+                Axis containing the performance plot.
+        """
+        plt.figure(**figure_kwargs)
+
+        imp_methods = list(self.report_df["strategy"])
```

Comment:
let's take the report using self.compute. this way we also make sure it is fitted first.

Reply:
Done.